dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/bundle/persister/DefaultTrackerObjectsDeletionService.java

```diff
@@ -79,30 +80,32 @@ public class DefaultTrackerObjectsDeletionService implements TrackerObjectDeleti
   private final ProgramNotificationInstanceService programNotificationInstanceService;
 
   @Override
-  public TrackerTypeReport deleteEnrollments(List<String> enrollments) throws NotFoundException {
+  public TrackerTypeReport deleteEnrollments(List<UID> enrollments) throws NotFoundException {
     UserInfoSnapshot userInfoSnapshot = UserInfoSnapshot.from(getCurrentUserDetails());
     TrackerTypeReport typeReport = new TrackerTypeReport(TrackerType.ENROLLMENT);
 
-    for (String uid : enrollments) {
-      Entity objectReport = new Entity(TrackerType.ENROLLMENT, uid);
+    for (UID uid : enrollments) {
+      Entity objectReport = new Entity(TrackerType.ENROLLMENT, uid.getValue());
 
-      Enrollment enrollment = manager.get(Enrollment.class, uid);
+      Enrollment enrollment = manager.get(Enrollment.class, uid.getValue());
```

Comment:
Can you add a `manager.get(class, UID)`? This will save us a lot of changed lines